src/Modal/Modal.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { ReactNode, FC } from 'react'
+import styled, { css } from 'styled-components'
+import { theme } from '../theme'
+
+import { Box } from '../Box'
+import { Icon } from '../Icon'
+import { Text } from '../Text'
+
+interface IModal {
+  // showModal state
+  showModal: boolean
+}
+
+type ModalProps = {
+  title?: string,
+  body?: string,
+  children?: ReactNode,
+  showModal?: boolean,
+  handleClick: () => void
+}
+
+export const Modal: FC<ModalProps> = ({
+  title = '',
+  body = '',
+  children = '',
+  showModal = false,
+  handleClick,
+}) => {
+
+  return (
+    <Wrapper showModal={showModal}>
+      <Overlay />
+      <Container>
+        <Box flex alignItems="center" justifyContent="space-between">
+          <Text tag="h2" typo="header-small" align="left">{title}</Text>
+          <IconContainer onClick={handleClick}>
```

Comment:
Does Jason want us to use the new icon button with this modal (seen here https://www.figma.com/file/FqSaizGOyOzXFZNmZ4X0LGMn/%F0%9F%8D%AC-S-mores?node-id=2790%3A0)?

Reply:
Discussing now 😄